### ISSUE TITLE
After 18.10 upgrade, new host or service downtime date defaults to yesterday in web UI

### ISSUE THREAD

**Issue by jonmilby** [centreon/centreon#6986]
---------------------------------------------------
BUG REPORT INFORMATION
---------------------------------------------------

**Centreon Web version**: 18.10.0

**Centreon Engine version**: 18.10.0

**Centreon Broker version**: 18.10.0

**OS**: CentOS 7.5

**Additional environment details (AWS, VirtualBox, physical, etc.):** Virtual (VMware)

**Steps to reproduce the issue:**
1. Upgraded from Centreon 3.4 to 18.10.0
2. Open Web UI, select hosts or services and add downtime from drop-down menu
3. Check default date/time in dialog box.

**Describe the results you received:** Start time is the current time, but start and end date are yesterday's date.

**Describe the results you expected:** Current date/time should be populated in start time, as was the case in previous versions of Centreon.

**Additional information you think important (e.g. issue happens only occasionally):** Host date/time is correct, system time zone is defined correctly and is the same as date.timezone in /etc/opt/rh/rh-php71/php.d/php-timezone.ini and /etc/php.ini (America/New_York).

**Comment by lpinsivy** [comment]
Hi,

I can reproduce this issue. All users have the same issue?

<img width="940" alt="downtime" src="https://user-images.githubusercontent.com/3351916/49434069-aab51180-f7b3-11e8-8400-9539f21111ad.png">


Regards,

**Comment by jonmilby** [comment]
Yes, the same issue occurs for all users in the system.  Only other difference between my screen and yours is the default downtime duration.
![screen shot 2018-12-04 at 11 44 07 am](https://user-images.githubusercontent.com/3143350/49458020-f5d22300-f7b9-11e8-9a0e-9fe71bb4703d.png)

**Comment by JonBogan** [comment]
I prefer to not do a pull request but will provide the fix here.

In `include/monitoring/downtime/AddDowntime.php` add the following at line 321:

    $data["start"] = $centreonGMT->getDate("m/d/Y", time(), $gmt);
    $data["end"] = $centreonGMT->getDate("m/d/Y", time() + 7200, $gmt);

In `include/monitoring/external_cmd/popup/massive_downtime.php` add the following at line 144:

        "start" => $centreonGMT->getDate("m/d/Y", time(), $gmt),
        "end" => $centreonGMT->getDate("m/d/Y", time() + 7200, $gmt),

These lines were in previous versions.

**Comment by jonmilby** [comment]
Thanks for the fix, @JonBogan - that appears to fix the issue on my install, which is now version 18.10.2!  @lpinsivy - can you review for a possible patch to future versions?

**Comment by adr-mo** [comment]
Hi @jonmilby 

This will be fixed in the **18.10.x and 19.04.x**

Regards,

**Comment by jonmilby** [comment]
Many thanks, @adr-mo !  Will be much easier not to have to patch after every update.

**Comment by roysha1** [comment]
Hi,

I got the same situation after upgrading from 3.4 to 19.10. Only some users are experiencing the issue described.

Per comments above, fix should already be in 19, and the issue still open.

OS: CentOS 7.5

Additional environment details (AWS, VirtualBox, physical, etc.): Virtual (KVM)

**Comment by jonmilby** [comment]
Yes, the bug actually reappeared for me in the next minor release after they initially patched it.  I had to resort back to manually patching every time I update Centreon.

**Comment by lpinsivy** [comment]
Hi

A fix have been made and changed the previous behaviour: https://github.com/centreon/centreon/commit/f06c69498fbf3bb1860c36a2b4afaf4e1d8c9cbd#diff-d73252cdaa3d27a76bf712a18addbabc

What is the problem you are facing now?

**Comment by jonmilby** [comment]
It looks like that fix has been reverted in the current master branch.  At least the portion from lines 320-323 is no longer the same.  Each time I upgrade Centreon versions, including to 19.04.8 today, I have to repatch AddDowntime.php to fix the original issue - new downtime UI defaults to yesterday.
